openapi/components/schemas/RichBillingPortal.yaml

```
@@ -0,0 +1,140 @@
+type: object
+required:
+  - name
+  - recurringOptions
+  - levels
+  - solutions
+properties:
+  id:
+    type: string
+    description: ID of the billing portal.
+    readOnly: true
+    maxLength: 50
+    example: bill_prt_0YV7K5TYV5D1P9ZNKDT39KZC3C
+  name:
+    type: string
+    description: Billing portal name.
+  recurringOptions:
+    type: array
+    description: Recurring options for the billing portal. For example, it can have `monthly` and `yearly` variants.
+    items:
+      type: object
+      properties:
+        id:
+          type: string
+          description: |-
+            ID of a recurring options.
+            It must be unique between other recurring options in the same billing portal.
+        name:
+          type: string
+          description: Name of recurring options for the billing portal.
+          example: monthly
+        recurringInterval:
+          type: object
+          description: Recurring interval settings.
+          properties:
+            unit:
+              description: Unit of time.
+              type: string
+              enum:
+                - day
+                - week
+                - month
+                - year
+            length:
+              description: Length of time.
+              type: integer
+              minimum: 1
+  levels:
+    type: array
+    description: Level options for billing portal solution.
+    items:
+      type: object
+      properties:
+        id:
+          type: string
+          description: |-
+            ID of a level.
+            It must be unique between other levels in the same billing portal.
```

suggestion: for readability
```suggestion
            This value must be unique to other levels in the same billing portal.
```